CryptKeeperZK/crypt-keeper-extension
# fix: detect metamask wallet installed

## Explanation

This PR fixes the state of extension when there is no metamask installed.

Details are below:
- [x] Replace `metamask-extension-provider` package with `@metamask/providers`
- [x] Check if wallet is injected
- [x] Fix loading state
- [x] Add metamask url and redirect to page

## More Information

Closes #242 

## Screenshots/Screencaps

### Before

![image](https://user-images.githubusercontent.com/14254374/230174772-6a6776d4-545d-410d-a9d3-62b19aa0f998.png)

### After

![image](https://user-images.githubusercontent.com/14254374/230174872-3c790b43-de77-49ed-ab6c-7b855e7f9a7b.png)

## Manual Testing Steps

1. Disabled metamask extension
2. Login into crypt keeper
3. Click connect icon and go to install page
4. Check there is no step 3 if metamask is enabled

## Pre-Merge Checklist


### + If there are functional changes:


> PR template source from [github.com/MetaMask](https://github.com/MetaMask)
